### PR TITLE
fix: Don't export Insights that are deleted

### LINKED ISSUE #11374: Exporting of dashboards or insights fail when a related Insight is deleted
## Bug description

When we delete Insights, they are soft deleted (marked as deleted=true) but this doesn't trigger some cascade-like deletions we would expect - specifically, Subscriptions and DashboardTiles.

This means that:
1. Subscriptions for deleted Insights still run but fail to generate the image as the Insight is deleted
2. Subscriptions for dashboards with a deleted Insight fail as they are trying to to export something that is gone.
3. If one image fails the whole Slack message fails (I think) as it doesn't accept messages with images that can't be loaded.

*Please describe.*  
*If this affects the front-end, screenshots would be of great help.*  

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Closes https://github.com/PostHog/posthog/issues/11374

## Changes

* Filters out subscription  scheduling of deleted insights or dashboards
* Filters out dashboard tiles to be exported if the insight is deleted

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Tests!

### CODE REVIEW

File: ee/tasks/test/subscriptions/test_subscriptions.py [hunk 1]
```diff
@@ -65,6 +66,39 @@ def test_subscription_delivery_scheduling(
...
+        mock_gen_assets: MagicMock,
+        mock_send_email: MagicMock,
+        mock_send_slack: MagicMock,
+    ) -> None:
+        self.insight.deleted = True
+        self.insight.save()
+        self.dashboard.deleted = True
+        self.dashboard.save()
+
+        create_subscription(
```
Comment (pauldambra) [inline ee/tasks/test/subscriptions/test_subscriptions.py:82]: it's a nit but should we subscribe before deleting? as a more realistic order of setup

**Review by pauldambra — APPROVED** [review]
teeny nit but otherwise 🚢

File: ee/tasks/test/subscriptions/test_subscriptions.py (hunk 1 above)
Comment (benjackwhite) [reply ee/tasks/test/subscriptions/test_subscriptions.py:82]: Good point

Comment (benjackwhite) [reply ee/tasks/test/subscriptions/test_subscriptions.py:77]: Done - still passes but might save us somehow in the future 🤯

**Pull request merged**